### PR TITLE
chore(file-system-api): drop search view

### PR DESCRIPTION
**Pull request by tlgimenes**
# What's the purpose of this PR?
This PR removes the search template from `gatsby-theme-store` and uses the Gatsby's File Sytem Route API for creating collection pages on the stores. Removing the search temaplate from `gatsby-theme-store` allowed us to do following breaking changes
1. (gatsby-source-vtex): Remove the staticPaths API
2. (gatsby-source-vtex); Remove Department sourcing
3. (gatsby-theme-store): Remove search template
4. (gatsby-theme-store): Drop createPage API all along

# Upgrading my store
To upgrade your store, you can use the changes made in our starter available at: https://github.com/vtex-sites/storecomponents.store/pull/1081.

TLDR: To have brand/cateogory pages, you need to create 3 pages with the following layout:
```
   | pages 
   -- | {StoreCollection.slug}
       -- | [...].tsx
       -- | index.tsx
   -- | s
       -- | [...].tsx
 ```
 
Now, a page will be created for each category/brand in your store. Also a `/s/:term` page will be created for full text searches.
You can use these newly created pages to implement the layout you desire

# How to test it?
This is just a code refactor and no visible feature should be changed. Make sure there are no regressions at:

https://github.com/vtex-sites/btglobal.store/pull/772
https://github.com/vtex-sites/marinbrasil.store/pull/585

### CODE REVIEW

File: packages/gatsby-theme-store/src/components/Search/Controls/index.tsx [hunk 1]
```diff
@@ -5,11 +5,9 @@ import type { FC } from 'react'
 
 import { SearchControlsFiltersButton } from './FiltersButton'
 import SearchControlsSelect from './SortSelect'
-import type { BrowserSearchPageQueryQuery } from '../../../templates/__generated__/BrowserSearchPageQuery.graphql'
-import type { ServerSearchPageQueryQuery } from '../../../templates/__generated__/ServerSearchPageQuery.graphql'
 
 interface Props {
-  data: BrowserSearchPageQueryQuery | ServerSearchPageQueryQuery
+  data: any // TODO
```
Comment (igorbrasileiro) [inline packages/gatsby-theme-store/src/components/Search/Controls/index.tsx:10]: This TODO should be here?

File: packages/gatsby-theme-store/src/sdk/search/Provider.tsx [hunk 2]
```diff
@@ -10,10 +10,8 @@ import React, { createContext, useMemo, useState } from 'react'
 import type { FC } from 'react'
 import type { SearchParamsState } from '@vtex/store-sdk'
 
-import type { ServerSearchPageQueryQuery } from '../../templates/__generated__/ServerSearchPageQuery.graphql'
-import type { BrowserSearchPageQueryQuery } from '../../templates/__generated__/BrowserSearchPageQuery.graphql'
-
-type SearchQuery = ServerSearchPageQueryQuery | BrowserSearchPageQueryQuery
+// TODO. I think we should remove data from this context once all componets are moved to the stores
```
Comment (igorbrasileiro) [inline packages/gatsby-theme-store/src/sdk/search/Provider.tsx:13]: This TODO should be here?

**Review by igorbrasileiro — APPROVED** [review]

**Comment by igorbrasileiro** [comment]
This PR is too large, maybe my review was not the best I could do.

File: packages/gatsby-theme-store/src/sdk/search/Provider.tsx (hunk 2 above)
Comment (tlgimenes) [reply packages/gatsby-theme-store/src/sdk/search/Provider.tsx:13]: Yes, this is related to #882

File: packages/gatsby-theme-store/src/components/Search/Controls/index.tsx (hunk 1 above)
Comment (tlgimenes) [reply packages/gatsby-theme-store/src/components/Search/Controls/index.tsx:10]: Yes, this is fixed on #882 since this will be moved to the stores

**Pull request merged**